server/chat-plugins/calculator.ts

```diff
@@ -132,17 +139,51 @@ export const commands: ChatCommands = {
 	math: "calculate",
 	calculate(target, room, user) {
 		if (!target) return this.parse('/help calculate');
+		let base = 10;
+		if (target.includes('0x')) {
+			base = 16;
+		} else if (target.includes('0o')) {
+			base = 8;
+		} else if (target.includes('0b')) {
+			base = 2;
+		}
+
+		const baseMatchResult = target.match(/\bin\s+([a-zA-Z]+)\b/);
+		if (baseMatchResult) {
+			switch (toID(baseMatchResult[1])) {
+			case 'decimal': case 'dec': base = 10; break;
+			case 'hexadecimal': case 'hex': base = 16; break;
+			case 'octal': case 'oct': base = 8; break;
+			case 'binary': case 'bin': base = 2; break;
+			default:
+				return this.errorReply(`Unrecognized base "${baseMatchResult[1]}". Valid options are binary or bin, octal or oct, decimal or dec, and hexadecimal or hex.`);
+			}
+		}
+		const expression = target.replace(/\bin\s+([a-zA-Z]+)\b/g, '').trim();
+
 		if (!this.runBroadcast()) return;
 		try {
-			const result = solveRPN(parseMathematicalExpression(target));
-			this.sendReplyBox(Chat.html`${target}<br />= <strong>${Chat.stringify(result)}</strong>`);
+			const result = solveRPN(parseMathematicalExpression(expression));
+			let baseResult = ``;
+			if (result) {
+				baseResult = `${BASE_PREFIXES[base]}${result.toString(base).toUpperCase()}`;
+				if (baseResult === result.toString()) baseResult = '';
```

Comment:
```suggestion
				if (baseResult === result.toString() || baseResult === expression) {
					baseResult = '';
				}
```

There's no need to display `0x10 = 0x10 = 16`